client/webserver/webserver.go

```diff
@@ -564,6 +571,21 @@ func (s *WebServer) Connect(ctx context.Context) (*sync.WaitGroup, error) {
 	return &wg, nil
 }
 
+// prepareAddr prepares the listening address in case a :0 was provided.
+func prepareAddr(addr net.Addr) (string, bool) {
+	// If the IP is unspecified, defualt to `127.0.0.1`. This is a workaround
+	// for an issue where all ip addresses other than exactly 127.0.0.1 will
+	// always fail to match when used in CSP directives. See:
+	// https://w3c.github.io/webappsec-csp/#match-hosts.
+	defaultIP := net.IP{127, 0, 0, 1}
+	tcpAddr, ok := addr.(*net.TCPAddr)
+	if ok && (tcpAddr.IP.IsUnspecified() || tcpAddr.IP.Equal(net.IPv6loopback) || tcpAddr.IP.Equal(defaultIP)) {
```

Comment:
What's this `tcpAddr.IP.Equal(defaultIP)` condition for?

Reply:
Was aiming at a one-liner, but maybe I should add a few more lines.

Reply:
OK, I suppose it's fine.  Was just confusing.  Looked like a bug.